src/client/gpm_import_and_canon_name.c

```diff
@@ -275,9 +289,29 @@ OM_uint32 gpm_canonicalize_name(OM_uint32 *minor_status,
         goto done;
     }
 
-    /* steal output_name */
-    *output_name = res->output_name;
-    res->output_name = NULL;
+    if (localname) {
```

Comment:
I think this it would be more clear to invert some of these checks.  Shape would be something like:

```C
if (!localname) {
    /* steal output_name */
    *(gss_name **)output_name = res->output_name;
    res->output_name = NULL;
    goto done;
}

gp_options_find(val, res->options, ...);
if (!val) {
    ret = ENOTSUP;
    goto done;
}

name = malloc(...);
if (!name) {
    ret = ENOMEM;
    goto done;
}

/* steal value */
*(gssx_name **)output_name = res->output_name;
res->output_name = NULL;

done:
    ...
```

(If nothing else, this makes it easier to see that in no case do we need to worry about releasing `name`.)

Reply:
alright